src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2Serializer.cs

```diff
@@ -302,6 +516,28 @@ public virtual Saml2Assertion ReadAssertion(XmlReader reader)
 
                 throw LogReadException(LogMessages.IDX13102, ex, Saml2Constants.Elements.Assertion, ex);
             }
+        }   
+
+        internal Saml2EncryptedAssertion ReadEncryptedAssertion(string assertion)
+        {
+            using (var stringReader = new StringReader(assertion))
+            {
+                var settings = new XmlReaderSettings { DtdProcessing = DtdProcessing.Prohibit };
+#if NET45 || NET451
+                settings.XmlResolver = null;
+#endif
```

Comment:
as mentioned, a utility method for this and perhaps even for the variations of readers.

Reply:
agree - same settings are created more than once throughout the project.

Reply:
addressed with: bfaa792c407991dd698f5922841e4bfa03884662